integration-cli/docker_cli_build_test.go

```diff
@@ -3675,15 +3675,35 @@ func TestBuildDockerignore(t *testing.T) {
 		RUN [[ ! -e /bla/src/_vendor ]]
 		RUN [[ ! -e /bla/.gitignore ]]
 		RUN [[ ! -e /bla/README.md ]]
+		RUN [[ ! -e /bla/dir/dir/f1 ]]
```

Comment:
Let's add your integration test that you removed.  I really think we need to test the heck out of this. In particular, that new match() logic with wildcards and multi-level dir patterns, etc...   Either here or in the unit tests.